examples/watch_resources/watch_test.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package watch_resources
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestWatchForResources(t *testing.T) {
+	watchFeature := features.New("test watcher").WithLabel("env", "dev").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			cl, err := cfg.NewClient()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			dep := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: "watch-dep", Namespace: cfg.Namespace()},
+			}
+
+			// watch for the deployment and triger action based on the event recieved.
+			go cl.Resources().Watch(ctx, &appsv1.DeploymentList{}, resources.WithFieldSelector(labels.FormatLabels(map[string]string{"metadata.name": dep.Name}))).
```

Comment:
It would be great to have an example of `Stop` handling too. Also, would it be a good idea to document and say we should explicitly invoke the `Stop` after the watch for the feature is done ? Otherwise this will leak a bunch of routines over the span of tests right ?

Reply:
Yes, thanks for pointing out this. Added an example how to use Stop().